onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -404,18 +415,44 @@ def collect_data(self, data_reader: CalibrationDataReader):
             raise TypeError(f"compute_data must return a TensorsData not {type(t)}.")
         self.clear_collected_data()
 
+
     def merge_range(self, old_range, new_range):
         if not old_range:
             return new_range
 
         for key, value in old_range.items():
+            # Check if stride is used and adjust merging strategy accordingly
+            if self.stride:
```

Comment:
Is there a reason why we need to check `self.stride`? It seems to me that just always using your new merging strategy would always work fine. I could be wrong though.